app/templates/_base_page.html

```diff
@@ -29,6 +29,42 @@
   {% block breadcrumb %}{% endblock %}
   <div id="wrapper">
     <main id="content" role="main">
+      {% with messages = get_flashed_messages(with_categories=true) %}
+        {% if messages %}
+          {% for category, message in messages %}
+            {% if category == 'track-page-view' %}
+              <div data-analytics="trackPageView" data-url="{{ message }}"></div>
+            {% elif category == 'flag' %}
+              {# these were originally messages with a function similar to the above "track-page-view" ones (analytics) but #}
+              {# they did not know about their actual virtual page view url, expecting the receiving page to know the       #}
+              {# corresponding url for a particular symbolic "message".                                                     #}
+              {# TODO these can be got rid of as soon as the app generating each flag flash message is converted to using   #}
+              {# new "track-page-view" mechanism                                                                            #}
+              {% if message == "account-created" %}
+                <div data-analytics="trackPageView" data-url="/suppliers?account-created=true"></div>
```

Comment:
Looking at the User FE: we have the redirect destination (used here in `data-url`) at the point where we trigger the flash message. So we should just be able to do:

```
redirect_path = '/suppliers' if current_user.role == 'supplier' else .... # etc
flash("{}?account-created=true".format(redirect_path), category="track-page-view")
return redirect(redirect_path)
```
And the path will be slotted as the `{{ message }}` in the `trackPageView` above.

Reply:
Well ... if we were certain we could get the that user FE change merged (and released!) before this, we could just forget all the backwards compat...

Reply:
🎉 https://github.com/alphagov/digitalmarketplace-user-frontend/pull/50

Reply:
Cool, but as discussed I think I still need to merge this ugliness to allow a clean transition...

Reply:
Agreed. We shouldn't forget to clean up the equivalent Buyer FE one as well afterwards.